### PR TITLE
Price Comparison and Checker

### LINKED ISSUE #545: Price Comparision and Checker
# Aim

- The aim of this python script would be to automatically search for an item across Amazon.com, Flipkart and other online shopping websites to find the cheapest price available.

# Details

- User can enter the item to be searched for.
- The script displays all possible matches in ascending order of prices, in the form of Site - Link - Cost
- It also displays the best/cheapest price that is available.
- Will start with Amazon and Flipkart first and then add other websites.

### PR DESCRIPTION
**Pull request by aashishah**
# Description

The python script automatically searches for an item across Amazon, Flipkart and other online shopping websites and after performing comparisons, finds the cheapest price available.

## Have you read the [Contributing Guidelines on Pull Requests](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/CONTRIBUTING.md)?

- [x] Yes
- [ ] No

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines(Clean Code) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have created a helpful and easy to understand `README.md`
- [x] My documentation follows [`Template for README.md`](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/Template%20for%20README.md)
- [x] My changes generate no new warnings
- [x] I have added tests/screenshots(if any) that prove my fix is effective or that my feature works.

### CODE REVIEW

File: Price Comparison and Checker/Readme.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+- Ensure you have Python installed in your system.
+- Simply run the program titled "price_comparison.py" or execute on Command Prompt/Terminal.
+- Next, You will be prompted to enter the item to be searched for and upon doing so, the program will list 5 products from all available websites (currently Amazon, Flipkart and Snapdeal) and give you the best price from all those available.
+
+## Output
+![screenshot](https://user-images.githubusercontent.com/49470807/111500289-d6344e00-8769-11eb-99ea-bf24db4e1b41.PNG)
+
+## Author(s)
+
+##[Aashi Shah](https://github.com/aashishah)
```
Comment (kaustubhgupta) [inline Price Comparison and Checker/Readme.md:23]:
```suggestion
## [Aashi Shah](https://github.com/aashishah)
```

File: Price Comparison and Checker/Requirements.txt [hunk 2]
```diff
@@ -0,0 +1,4 @@
+##Modules used in the program
```
Comment (kaustubhgupta) [inline Price Comparison and Checker/Requirements.txt:1]: This line is not needed. Also, rename the file to requirements.txt

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
The code is working fine just some minor changes:

File: Price Comparison and Checker/price_comparison.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+	converted_price = [] 
+	for i in price: 
+		if i.isdigit():
+			converted_price.append(i) #Extracting only digits
+
+	converted_price = int("".join(converted_price)) #Converting the string price to integer for comparison
+	return converted_price
+
+
+item = input("Enter the item you would like to search for: ")
```
Comment (antrikshmisri) [inline Price Comparison and Checker/price_comparison.py:96]:
```suggestion
if __name__ == "__main__":
    item = input("Enter the item you would like to search for: ")
```

Comment (antrikshmisri) [reply Price Comparison and Checker/price_comparison.py:96]: wrap everything from line `96` to `104` in this if statement. If you try to import this file in any other file the functions in line `97`,`98`,`99` will run immediately if not wrapped in this if statement.

**Review by antrikshmisri — CHANGES_REQUESTED** [review]

File: Price Comparison and Checker/price_comparison.py (hunk 3 above)
Comment (antrikshmisri) [inline Price Comparison and Checker/price_comparison.py:96]:
```suggestion
if __name__ == "__main__":
    item = input("Enter the item you would like to search for: ")
```

Comment (antrikshmisri) [reply Price Comparison and Checker/price_comparison.py:96]: wrap everything from line `96` to `104` in this if statement. If you try to import this file in any other file the functions in line `97`,`98`,`99` will run immediately if not wrapped in this if statement.

**Review by antrikshmisri — CHANGES_REQUESTED** [review]

File: Price Comparison and Checker/Readme.md (hunk 1 above)
Comment (aashishah) [reply Price Comparison and Checker/Readme.md]: Made the changes. Please check.

**Comment by aashishah** [comment]
@kaustubhgupta I've pushed the updated code. Kindly check.

**Review by kaustubhgupta — APPROVED** [review]
LGTM 💯

**Review by antrikshmisri — APPROVED** [review]
LGTM

**Review by santushtisharma10 — APPROVED** [review]
@aashishah 💯

**Review by avinashkranjan — APPROVED** [review]
LGTM..🚀

**Pull request merged**